tests/test_types.py

```diff
@@ -316,22 +316,75 @@ class Config:
         max_anystr_length = 10
 
 
+# class BoolCastable:
+#     def __init__(self, x: int) -> None:
+#         self.x = x
+#
+#     def __bool__(self) -> bool:
+#         return self.x > 0
+#
+#
+# class BoolCastableModel(BaseModel):
+#     x: int
+#
+#     def __bool__(self) -> bool:
+#         return self.x > 0
+#
+# class NonBoolCastableModel(BaseModel):
+#     x: int
+
```

Comment:
I will replace these commented classes with appropriate alternatives (and alter/remove references in the non-test code) once we have a consensus on `__bool__` handling.